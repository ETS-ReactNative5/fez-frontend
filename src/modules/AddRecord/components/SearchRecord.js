import React from 'react';
import PropTypes from 'prop-types';

// forms & custom components
import {PublicationSearchForm} from 'modules/PublicationSearchForm';

import {locale, PATHS} from 'config';

export default class SearchRecord extends React.Component {
    static propTypes = {
        actions: PropTypes.object,
        history: PropTypes.object.isRequired
    };

    constructor(props) {
        super(props);
    }

    _performSearch = (values) => {
        this.props.actions.searchPublications(values.get('searchQuery'));
        this.props.history.push(PATHS.records.add.searchResults);
    };

    render() {
        const txt = locale.pages.addRecord;
        return (
            <PublicationSearchForm locale={txt.step1} onSubmit={this._performSearch}/>
        );
    }
}
