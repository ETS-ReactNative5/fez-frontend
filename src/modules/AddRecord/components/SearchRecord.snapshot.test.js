import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import SearchRecord from './SearchRecord';

function setup(values) {
    return shallow(<SearchRecord {...values}/>);
}

describe('Search record', () => {
    it('should render stepper and a publication search form', () => {
        const wrapper = setup({history: {}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
