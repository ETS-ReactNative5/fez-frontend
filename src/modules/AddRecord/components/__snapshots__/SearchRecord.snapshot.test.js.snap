// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search record should render stepper and a publication search form 1`] = `
<Connect(ReduxForm)
  locale={
    Object {
      "fieldLabels": Object {
        "search": "Enter DOI, Pubmed Id or Title",
      },
      "help": Object {
        "buttonLabel": "Ok",
        "text": "Help about search....",
        "title": "Search for your publication",
      },
      "submit": "Search",
      "text": "Enter either the publication DOI (e.g. 10.1163/9789004326828), Pubmed Id (e.g. 28131963) or the title of the publication. This will allow us to check whether the record is already in eSpace or is available from another source.",
      "title": "Search for your publication",
    }
  }
  onSubmit={[Function]}
/>
`;
